fuzzing/private/oss_fuzz/package.bzl

```diff
@@ -44,6 +44,32 @@ def _oss_fuzz_package_impl(ctx):
                 ln -s "$(pwd)/{dictionary_path}" "$STAGING_DIR/{base_name}.dict"
             fi
             tar -chf "{output}" -C "$STAGING_DIR" .
+            # The runfiles listed in this variable are made available to
+            # fuzz tests by OSS-Fuzz in the top-level project $OUT directory.
+            # We replace the Bazel-generated runfile with a symlink to that
+            # top-level file.
+            declare -r RUNFILES_TO_PATCH=" \
+                jazzer/agent/jazzer_agent_deploy.jar \
+                jazzer/driver/jazzer_driver \
+                jazzer/driver/jazzer_driver_asan \
+                "
+            for RUNFILE in $RUNFILES_TO_PATCH; do
+                RUNFILE_PATH="{base_name}.runfiles/$RUNFILE"
+                [[ -e "$STAGING_DIR/$RUNFILE_PATH" ]] || continue
+                tar --delete -f "{output}" "./$RUNFILE_PATH"
+                # Generate a path that when resolved relative to RUNFILE_PATH's
+                # parent directory points to a top-level OSS-Fuzz binary of the
+                # same basename.
+                RUNFILE_RELATIVE_PATH_TO_GLOBAL="$(echo $RUNFILE_PATH | sed 's|[^/]*/|../|g')"
```

Comment:
What is a top-level OSS-Fuzz binary? Is that an absolute path?

I'm not sure such paths would actually be available to the Bazel rule during building, since all builds are hermetic by default and can only access the explicitly-mentioned input files.

Reply:
What I mean by "top-level OSS-Fuzz binary" here is a binary made available under a fixed name directly in `$OUT/` by OSS-Fuzz `compile` script. This currently applies to `jazzer_driver`, `jazzer_driver_asan`, and `jazzer_agent_deploy.jar` if the language of the project is set to `jvm`.

These paths would not need to be available to Bazel during building the way this PR is currently designed: In the final step, the actual runfile version of those files built with Bazel are replaced by relative symlinks that resolve to the corresponding binaries in `$OUT/`. Since the resulting .tar archives are extracted directly into `$OUT/`, the relative path from the runfile `jazzer_driver` to `$OUT/jazzer_driver` is fixed and can be computed up front, which is what this line is supposed to do. Since the symlink is dangling while in the tar and never resolved (note the lack of `-h` in the last `tar` command), this works fine in the sandbox.

I'm not sure I don't miss anything here, but I believe the only assumption this part of the script makes is that the target `@jazzer//driver:jazzer_driver` in any Bazel project that uses `rules_fuzzing` should always be replaced with the OSS-Fuzz provided `jazzer_driver` (and e.g. is not something else entirely that just happens to have this label).